c-deps/BUILD.bazel

```diff
@@ -11,7 +11,16 @@ configure_make(
         "--enable-prof",
     ] + select({
         "@io_bazel_rules_go//go/platform:windows": ["--host=x86_64-w64-mingw32"],
-        "@io_bazel_rules_go//go/platform:darwin": ["--host=x86_64-apple-darwin19"],
+        "@io_bazel_rules_go//go/platform:darwin_amd64": ["--host=x86_64-apple-darwin19"],
+        "@io_bazel_rules_go//go/platform:darwin_arm64": ["--host=aarch64-apple-darwin19"],
+        # NB: Normally host detection is handled by configure, but the version
+        # of jemalloc we have vendored is pretty ancient and can't handle some
+        # of the newer M1 Macs. This arm of the select() can probably be deleted
+        # when we bump the vendored version.
+        "//build/toolchains:is_dev_macos_arm64": [
```

Comment:
Hmmm, hmmm. So I think this is a good patch to get things working until a proper fix can come in - this will hopefully be fixed properly with the jemalloc upgrade.

Just making a note that probably the long-term healthier way to do this would be to prefix the configure command with a script that checks for known unsupported execution and target platforms and does the relevant substitution on them before passing them as explicit --host and --build flags.

Reply:
I think that the jemalloc upgrade will indeed render this patch unnecessary.

> Just making a note that probably the long-term healthier way to do this would be to prefix the configure command with a script that checks for known unsupported execution and target platforms and does the relevant substitution on them before passing them as explicit --host and --build flags.

I agree. The whole `c-deps` situation is very dubious right now, hopefully we can take a good pass and make this all a little more robust.